### ISSUE TITLE
Vitest error with react-querybuilder on tests

### ISSUE THREAD

**Issue by balboFK** [react-querybuilder/react-querybuilder#270]
Hello everyone!
I'm migrating my tests from _jest_ to _vitest_ and one of my Components is a custom implementation of the react-querybuilder.

When I was using jest and version `3.12.1`, everything ran fine as allways.
Now I seen to have landed on a double header:

- migrating from jest to vitest
- migrating 'react-querybuilder' from `3.12.1` to `4.1.2`

So the error I get when running the tests now is the bellow:
```
Error: Failed to resolve entry for package "react-querybuilder". The package may have incorrect main/module/exports specified in its package.json: Failed to resolve entry for package "react-querybuilder". The package may have incorrect main/module/exports specified in its package.json.
 ❯ packageEntryFailure node_modules/vite/dist/node/chunks/dep-0351185a.js:30629:11
 ❯ resolvePackageEntry node_modules/vite/dist/node/chunks/dep-0351185a.js:30625:9
 ❯ tryNodeResolve node_modules/vite/dist/node/chunks/dep-0351185a.js:30440:20
 ❯ Context.resolveId node_modules/vite/dist/node/chunks/dep-0351185a.js:30257:28
 ❯ Object.resolveId node_modules/vite/dist/node/chunks/dep-0351185a.js:36922:55
 ❯ processTicksAndRejections ../../../../../internal/process/task_queues.js:95:5
 ❯ async TransformContext.resolve node_modules/vite/dist/node/chunks/dep-0351185a.js:36721:23
 ❯ async normalizeUrl node_modules/vite/dist/node/chunks/dep-0351185a.js:73692:34
 ❯ async TransformContext.transform node_modules/vite/dist/node/chunks/dep-0351185a.js:73834:57
 ❯ async Object.transform node_modules/vite/dist/node/chunks/dep-0351185a.js:36985:30
```

Any heading of what might be the issue on this? By the leaning of the error it seem like some mistaken entry got lost on module/exports, but I don't know exactly what could be.

EQ: @jakeboone02 do you plan on migrating to vitest aswell? I would like to hear your opinion on it since  v4.0 is using Vite now.

**Comment by jakeboone02** [comment]
Hey, @balboFK, can you create an minimal example repo that demonstrates the issue you're having with 4.1.2? I haven't heard of any similar problems with that version, and it appears to work fine on our [CI codesandbox](https://codesandbox.io/s/oq8ts).

I just heard about vitest yesterday, actually. Looks promising and yes, I'll definitely consider it for this library since we're already using vite. I'm a little wary about this message on the [home page](https://vitest.dev/), though: "It's not recommended to migrate your current testing setups yet". I'll check it out for sure.

**Comment by balboFK** [comment]
Hey @jakeboone02 thanks for the reply.
Take a look at this :
https://stackblitz.com/edit/vitest-dev-vitest-ffvyhj?devtoolsheight=33&file=src/App.test.tsx

Just run `yarn test` and watch the console.

**Comment by balboFK** [comment]
I think I might have found what could be the issue.

On react-querybuilder package.json your main is exporting as 'dist/index.umd.js'
![image](https://user-images.githubusercontent.com/87539237/149979765-2f26050f-0a7f-485f-9c33-e71e1dbc0a29.png)

And on the folder for dist this file is named as 'index.cjs.js'
![image](https://user-images.githubusercontent.com/87539237/149979935-a38925c7-a420-4058-a624-737eaa6caff2.png)

Maybe thats it?

**Comment by jakeboone02** [comment]
Yeah, that must be it! The `umd` name was probably just left over from the previous parcel-based build process. I'll get a v4.1.3 out today to fix it.

**Comment by balboFK** [comment]
Thanks @jakeboone02! 
I'll wait for it and update the StackBlitz i just sent to see if fixes it.
I tried changing it locally on my machine, it fixed the error but new ones showed up, but it might be my fault.

**Comment by jakeboone02** [comment]
@balboFK - [v4.1.3](https://github.com/react-querybuilder/react-querybuilder/releases/tag/v4.1.3) has been released.

**Comment by balboFK** [comment]
I'll take a look at it tomorrow and update the Stackblitz to see if it's fixed!
If it is, i'll close this one.
Thanks!

**Comment by balboFK** [comment]
Hey @jakeboone02, sorry for the delay!
I was able to test it today and I found another issue.
I got the bellow error when testing my CustomQueryBuilder component that uses the react-querybuilder.

```
Error: Cannot have two HTML5 backends at the same time.
 ❯ HTML5BackendImpl.setup node_modules/react-dnd-html5-backend/dist/cjs/HTML5BackendImpl.js:484:15
 ❯ node_modules/dnd-core/dist/cjs/classes/DragDropManagerImpl.js:37:25
 ❯ Object.dispatch node_modules/redux/lib/redux.js:305:7
 ❯ HandlerRegistryImpl.addSource node_modules/dnd-core/dist/cjs/classes/HandlerRegistryImpl.js:113:18
 ❯ registerSource node_modules/react-dnd/dist/cjs/internals/registration.js:19:27
 ❯ registerDragSource node_modules/react-dnd/dist/cjs/hooks/useDrag/useRegisteredDragSource.js:36:59
 ❯ commitHookEffectListMount node_modules/react-dom/cjs/react-dom.development.js:20573:26
 ❯ commitLifeCycles node_modules/react-dom/cjs/react-dom.development.js:20634:11
 ❯ commitLayoutEffects node_modules/react-dom/cjs/react-dom.development.js:23426:7
 ❯ HTMLUnknownElement.callCallback node_modules/react-dom/cjs/react-dom.development.js:3945:14
```

I think it's maybe because we are both using react-dnd in our projects?
Also this CustomQueryBuilder is not using react-dnd anywhere in fact it's only dependencies are react-querybuilder.
It's also an isolated module on it's own test, so i'm kinda struggling to find out what's happening this time.

**Comment by jakeboone02** [comment]
I tried adding `react-dnd` to [this CodeSandbox](https://codesandbox.io/s/hardcore-shtern-23y5k?file=/src/App.tsx) which also has `react-querybuilder`, but it didn't seem to cause any problems.

**Comment by jakeboone02** [comment]
FYI, the "two HTML5 backends" issue should be resolved in the next version (v4.2.0). Watch #276 for progress.

**Comment by balboFK** [comment]
Hey @jakeboone02, thanks for the information! 
I was really loaded with stuff and forgot to reply to you.
I'll follow the progress on the issue you linked.
When it's fixed we can close this one aswell, or do we close it now?

**Comment by jakeboone02** [comment]
Yeah, I'll close this one since the original issue was resolved in v4.1.3.
